tests/common/util.rs

```diff
@@ -1191,26 +1221,613 @@ impl UCommand {
+    /// with [`Stdio::piped`] or else this method doesn't work as expected. `Stdio::piped` is the
+    /// current default when using [`UCommand::run_no_wait`]) without calling `set_stdin`. This
+    /// method stores a [`JoinHandle`] of the thread in which the writing to the child processes'
+    /// stdin is running. The associated thread is joined with the main process in the methods below
+    /// when exiting the child process.
+    ///
+    /// * [`UChild::wait`]
+    /// * [`UChild::wait_with_output`]
+    /// * [`UChild::pipe_in_and_wait`]
+    /// * [`UChild::pipe_in_and_wait_with_output`]
+    ///
+    /// Usually, there's no need to join manually but if needed, the [`UChild::join`] method can be
+    /// used .
+    ///
+    /// [`JoinHandle`]: std::thread::JoinHandle
+    pub fn pipe_in<T: Into<Vec<u8>>>(&mut self, content: T) -> &mut Self {
+        let ignore_stdin_write_error = self.ignore_stdin_write_error;
+        let content = content.into();
+        let stdin = self
+            .raw
+            .stdin
+            .take()
+            .expect("Could not pipe into child process. Was it set to Stdio::null()?");
+
+        let join_handle = thread::spawn(move || {
+            let mut writer = BufWriter::new(stdin);
+
+            match writer.write_all(&content).and_then(|_| writer.flush()) {
+                Err(error) if !ignore_stdin_write_error => Err(io::Error::new(
+                    io::ErrorKind::Other,
+                    format!("failed to write to stdin of child: {}", error),
+                )),
+                Ok(_) | Err(_) => Ok(()),
+            }
+        });
+
+        self.join_handle = Some(join_handle);
+        self
+    }
+
+    /// Call join on the thread created by [`UChild::pipe_in`] and if the thread is still running.
+    /// This method can be called multiple times but is a noop if already joined.
+    pub fn join_in(&mut self) -> &mut Self {
+        if let Some(join_handle) = self.join_handle.take() {
+            join_handle
+                .join()
+                .expect("Error joining with the piping stdin thread")
+                .unwrap();
+        }
+        self
+    }
+
+    /// Convenience method for [`UChild::pipe_in`] and then [`UChild::wait`]
+    pub fn pipe_in_and_wait<T: Into<Vec<u8>>>(mut self, content: T) -> CmdResult {
+        self.pipe_in(content);
+        self.wait().unwrap()
+    }
+
+    /// Convenience method for [`UChild::pipe_in`] and then [`UChild::wait_with_output`]
+    pub fn pipe_in_and_wait_with_output<T: Into<Vec<u8>>>(mut self, content: T) -> Output {
+        self.pipe_in(content);
+        self.wait_with_output().unwrap()
+    }
+
+    /// Write some bytes to the child process stdin. This function is meant for small data and
+    /// faking user input like typing a `yes` or `no`. This function blocks until all data is
+    /// written but can be used multiple times in contrast to [`UChild::pipe_in`].
```

Comment:
Excellent explanation, thanks!

A quick rustdoc tip, the first line is usually for a small summary, followed by a blank line a longer explanation. It will use the short summary in the pages for modules for example. I suppose though that we don't really need to worry about that in this case though because I don't even know how to run rustdoc for this file.

Reply:
> A quick rustdoc tip, the first line is usually for a small summary, followed by a blank line a longer explanation.

Thanks. It's a good idea to separate the first sentence from the rest, if the docs can be built or not. I had the test writers in mind and the docs popup in the IDE.